tests/Unit/DataStructures/Tensor/EagerMath/Test_Magnitude.cpp

```diff
@@ -261,6 +267,82 @@ void test_root_tags() {
   CHECK(Tags::Sqrt<Tag>::name() == "Sqrt(" + Tag::name() + ")");
   /// [sqrt_name]
 }
+
+void test_l2_norm_tag() {
+  constexpr size_t npts = 5;
+  MAKE_GENERATOR(generator);
```

Comment:
I don't think you should do a random-value test that re-implements parts of the code you want to test. I'd prefer to have hard-coded numbers and compare to other hard-coded numbers.

Reply:
I am a bit confused by what you mean when you say that the test _re-implements parts of the code you want to test_. 

The code to be tested computes the norm by going through (only) those tensor components that are stored on disk and uses their multiplicity, it never explicitly indexes the tensor whose norm is being taken. The test, on the other hand, accesses tensor components through explicit indexing. There is no loop over components in the test either. Which part of the tested code is being "re-implemented" in the test? The `square` and `add` operations?

Reply:
It re-implements the `PlusSquare` and `alg::accumulate`.

Generally, I'm worried when I see a test with no hard-coded values. And I don't see the benefit of random-value tests here since this implements no particularly complicated set of equations where random numbers could turn up edge-cases in the numerical implementation.

Reply:
I see what you mean, and have replaced all randomized tests with tests using hard-coded numbers for input/output.